lua/dapui/windows/layout.lua

```diff
@@ -82,10 +84,23 @@ function WindowLayout:_area_size()
   return 0
 end
 
-function WindowLayout:resize()
+function WindowLayout:resize(width, height)
   if not self:is_open() then
     return
   end
+
+  -- Detecting whether self.area_state.size is a float or int
+  local num, fraction = math.modf(self.area_state.size)
```

Comment:
Using the window size from above is unreliable, as it is affected by other open windows. Let's do something like this instead so it always uses the maximum size. (Also removed the `modf` usage as it's easier to read, no need to leave a comment)
```lua
  if self.area_state.size < 1 then
    if self.layout_type == "vertical" then
      local left = 1
      local right = vim.opt.columns:get()
      self.area_state.size = math.floor((right - left) * self.area_state.size)
    elseif self.layout_type == "horizontal" then
      local top = vim.opt.tabline:get() == "" and 0 or 1
      local bottom = vim.opt.lines:get() - (vim.opt.laststatus:get() > 0 and 2 or 1)
      self.area_state.size = math.floor((bottom - top) * self.area_state.size)
    else
      error('Unknown layout type')
    end
  end
```

Reply:
Yeah, that does sound better, replaced my logic with your snippet